pkg/sink/postgres.go

```diff
@@ -385,18 +391,28 @@ func (p *PGXSink) flushInsert(ctx context.Context) (err error) {
 	if len(batch) == 0 {
 		return nil
 	}
-	fields := len(batch[0])
-	rets := make([]int16, fields)
-	for i := 0; i < fields; i++ {
+
+	info, _ := p.schema.GetColumnInfo(p.inserts.Schema, p.inserts.Table)
+	cols, filtered := info.Filter(batch[0], func(i decode.ColumnInfo, field string) bool {
+		return !i.IsGenerated(field)
+	})
+
+	fCount := len(filtered)
+	rets := make([]int16, fCount)
+	for i := 0; i < fCount; i++ {
 		rets[i] = 1
 	}
-	fmts := make([]int16, fields*len(batch))
-	vals := make([][]byte, fields*len(batch))
-	oids := make([]uint32, fields*len(batch))
+	fmts := make([]int16, fCount*len(batch))
+	vals := make([][]byte, fCount*len(batch))
+	oids := make([]uint32, fCount*len(batch))
 	c := 0
 	for i := 0; i < len(batch); i++ {
-		for j := 0; j < fields; j++ {
+		for j := 0; j < len(batch[0]); j++ {
 			field := batch[i][j]
+			if !cols.Contains(field.Name) {
+				// skip the data since it's the generated column
+				continue
```

Comment:
no, since here we move the cursor `c` after setting the values, so skipping generated columns will not move the cursor and thus will not create holes.